lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -449,6 +457,15 @@ private boolean isCorruptedColumnIndex(ColumnIndex columnIndex)
         return false;
     }
 
+    public static long longExact(Number value)
+    {
+        if (value instanceof Double || value instanceof Float) {
+            throw new IllegalArgumentException("Can't convert inexact value to long: " + value.getClass().getName());
+        }
+
+        return value.longValue();
```

Comment:
`Number#longValue` doesn't realize the exactness guarantee given by the method name